src/debugger/breakpoints.rs

```diff
@@ -1,33 +1,330 @@
+
+            let test_inputs = testconst::tag_string_args();
+            let numeric_results: Vec<Option<usize>> = vec![
+                Some(0x808000),     // tag $808000
+                Some(50),           // tag 50
+                Some(0x80800A),     // tag $808000 + $0A
+                Some(0x808032),     // tag $808000 + 50
+                None,               // tag +$0A 
+                None,               // tag +50
+
+                /* Value before tag */
+                Some(0x808000),     // $808000 tag
+                Some(50),           // 50 tag
+                Some(0x80800A),     // $808000 + $0A tag
+                Some(0x808032),     // $808000 + 50 tag
+                None, // will fail  // $0A + tag
+                None, // will fail  // 50 + tag
+                Some(0x80800A),     // +$0A tag
+                Some(0x808032),     // +50 tag
+                
+                /* Other */
+                Some(0x808000),     // tag
+                None, // will fail, // tag + tag
+                None, // will fail, // tag + tag2
+                None, // will fail, // tag + tag2 tag3
+                None, // will fail  // tag3 tag + tag2
+            ];
+
+            for (test_input, expected_result) in zip(test_inputs, numeric_results){
+                if let Some(result) = expected_result {
+                    assert_eq!((), dbg_breakpoint_set(test_input, &mut test_debug, &mut test_vm).unwrap());
+                    assert!(test_debug.breakpoints.contains(&result));
+                    test_debug.breakpoints.clear();
+                    test_debug.tags.clear();
+                }
+                else{
+                    assert!(dbg_breakpoint_set(test_input, &mut test_debug, &mut test_vm).is_err());
+                }
             }
-            Err(e) => {
-                println!("{}", e);
+        }
+
+        #[test]
+        fn test_dbg_breakpoint_set_with_existing(){ 
+            let mut test_debug = DebuggerState::new();
+            let mut test_vm = VirtualMachine::new();
+
+            let test_inputs = testconst::tag_string_args();
+            let numeric_results: Vec<Option<usize>> = vec![
+                None,           // tag $808000
+                None,           // tag 50
+                None,           // tag $808000 + $0A
+                None,           // tag $808000 + 50
+                Some(0x80800A), // tag +$0A
+                Some(0x808032), // tag +50
+
+                /* Value before tag */
+                None,           // $808000 tag
+                None,           // 50 tag
+                None,           // $808000 + $0A tag
+                None,           // $808000 + 50 tag
+                Some(0x80800A), // $0A + tag
+                Some(0x808032), // 50 + tag
+                None,           // +$0A tag
+                None,           // +50 tag
+                
+                /* Other */
+                Some(0x808000), // tag
```

Comment:
This behavior is a little bit strange... May be worth re-evaluating whether to split tags off into individual categories per-command, but for now this is technically correct.